### PR TITLE
update sanity check to solve cidr_value and public_ips conflicts

### PR DESCRIPTION
**Pull request by matteofiordarancio**
I verified with a test that for compute_subnet_cidr public_ips has to be
set to true, so I added the check into the code.

### CODE REVIEW

File: cli/pcluster/config_sanity.py [hunk 1]
```diff
@@ -343,6 +343,11 @@ def validate(self, resource_type, resource_value):  # noqa: C901 FIXME
             ):
                 self.__fail(resource_type, "DNS Hostnames not enabled in %s" % resource_value)
         # VPC Subnet Id
+        elif resource_type == "VPC_COHERENCE":
```
Comment (enrico-usai) [inline cli/pcluster/config_sanity.py]: I think you are missing the cfnconfig.py file in this PR

Comment (matteofiordarancio) [reply cli/pcluster/config_sanity.py]: done

File: cli/pcluster/config_sanity.py [hunk 2]
```diff
@@ -343,6 +343,11 @@ def validate(self, resource_type, resource_value):  # noqa: C901 FIXME
...
+            cidr_value = resource_value[0]
+            public_ip = resource_value[1]
+            if cidr_value and public_ip is False:
+                self.__fail(resource_type, "compute_subnet_cidr needs use_public_ips to be true")
```
Comment (demartinofra) [inline cli/pcluster/config_sanity.py]: why is this required? is it always true? I'm wondering if it's a bug or a necessary requirement

File: cli/pcluster/cfnconfig.py [hunk 3]
```diff
@@ -354,6 +354,11 @@ def __init_vpc_parameters(self):
                     "VPC section [%s] used in [%s] section is not defined" % (vpc_section, self.__cluster_section)
                 )
 
+        # Check that cidr and public ips are not both set
+        cidr_value = self.__config.get(vpc_section, "compute_subnet_cidr", fallback=None)
+        public_ips = self.__config.getboolean(vpc_section,"use_public_ips", fallback=True)
+        self.__validate_resource("VPC_COHERENCE", (cidr_value, public_ips))
```
Comment (demartinofra) [inline cli/pcluster/cfnconfig.py]: if possible let's avoid adding more and more code to this __validate_resource function :)

Comment (matteofiordarancio) [reply cli/pcluster/cfnconfig.py]: how would you do that if not adding code there?

Comment (demartinofra) [reply cli/pcluster/cfnconfig.py]: You can extract the code under the `elif resource_type == "VPC_COHERENCE":` branch into a separate function (something like validate_vpc_coherence) and call that in place of the huge validate function.

Comment (matteofiordarancio) [reply cli/pcluster/cfnconfig.py]: Done, check if it is what you meant

File: cli/pcluster/cfnconfig.py [hunk 4]
```diff
@@ -354,6 +354,11 @@ def __init_vpc_parameters(self):
                     "VPC section [%s] used in [%s] section is not defined" % (vpc_section, self.__cluster_section)
                 )
 
+        # Check that cidr and public ips are not both set
+        cidr_value = self.__config.get(vpc_section, "compute_subnet_cidr", fallback=None)
+        public_ips = self.__config.getboolean(vpc_section, "use_public_ips", fallback=True)
+        ResourceValidator.validate_vpc_coherence(cidr_value, public_ips)
```
Comment (demartinofra) [inline cli/pcluster/cfnconfig.py]: we forgot an if self.__sanity_check:
